lib/protobuf/builder.ex

```diff
@@ -49,10 +49,18 @@ defmodule Protobuf.Builder do
         %{^field_name => value} when not is_nil(value) ->
           case props.field_props[props.field_tags[field_name]] do
             %FieldProps{embedded?: true, repeated?: true, type: type} ->
-              %{acc | field_name => Enum.map(value, &type.new/1)}
+              if type.transform_module() do
```

Comment:
Would it be too much of a performance hit if we moved this logic into `new` itself to prevent the duplication here? Also, if we don't expect `transform_module` to change at runtime, would it be possible do that check at compile time and generate different versions of `new`?

Reply:
Update: just realized it would probably make it harder to opt in to the original new-calling behavior that way 🤔